ui/src/app/components/apps-panel.spec.tsx

```diff
@@ -152,33 +152,38 @@ describe('AppsPanel', () => {
   // these tests assume that there are no Jupyter runtimes
   // so what these tests actually show is whether a GKE app is an ActiveApp
   test.each([
-    [AppStatus.RUNNING, true, true],
-    [AppStatus.STOPPED, true, true],
-    [AppStatus.STOPPING, true, true],
-    [AppStatus.STARTING, true, true],
-    [AppStatus.PROVISIONING, true, true],
-    [AppStatus.DELETING, true, true],
-    [AppStatus.STATUSUNSPECIFIED, true, true],
-    [AppStatus.ERROR, true, true],
+    [AppType.CROMWELL, AppStatus.RUNNING, true],
+    [AppType.CROMWELL, AppStatus.STOPPED, true],
+    [AppType.CROMWELL, AppStatus.STOPPING, true],
+    [AppType.CROMWELL, AppStatus.STARTING, true],
+    [AppType.CROMWELL, AppStatus.PROVISIONING, true],
+    [AppType.CROMWELL, AppStatus.DELETING, true],
+    [AppType.CROMWELL, AppStatus.STATUSUNSPECIFIED, true],
+    [AppType.CROMWELL, AppStatus.ERROR, true],
 
     // not visible [isVisible() = false]
-    [AppStatus.DELETED, false, true],
+    [AppType.CROMWELL, AppStatus.DELETED, false],
+
+    [AppType.CROMWELL, null, false],
 
-    [null, false, true],
+    // RStudio should have the same behavior.  Choose a few to demonstrate.
+
+    [AppType.RSTUDIO, AppStatus.RUNNING, true],
+    [AppType.RSTUDIO, AppStatus.DELETED, false],
```

Comment:
Another idea:

```
  // these tests assume that there are no Jupyter runtimes
  // so what these tests actually show is whether a GKE app is an ActiveApp
  const gkeAppTypes = [AppType.CROMWELL, AppType.RSTUDIO];

  describe.each(gkeAppTypes)('GKE App %s', (appType) => {
    test.each([
      [AppStatus.RUNNING, true],
      [AppStatus.STOPPED, true],
      [AppStatus.STOPPING, true],
      [AppStatus.STARTING, true],
      [AppStatus.PROVISIONING, true],
      [AppStatus.DELETING, true],
      [AppStatus.STATUSUNSPECIFIED, true],
      [AppStatus.ERROR, true],

      // not visible [isVisible() = false]
      [AppStatus.DELETED, false],

      [null, false],
    ])(
      'should render / not render ActiveApps and AvailableApps when status is %s',
      async (status, activeExpected) => {
        runtimeStub.runtime.status = RuntimeStatus.Deleted;
        appsStub.listAppsResponse = [{ status, appType }];

        const wrapper = await component();
        expect(wrapper.exists()).toBeTruthy();
        await waitOneTickAndUpdate(wrapper);

        // no Jupyter runtimes, so there is always at least one Available app (Jupyter)
        expect(
          findAvailableApps(wrapper, activeExpected).exists()
        ).toBeTruthy();

        expect(findActiveApps(wrapper).exists()).toBe(activeExpected);
      }
    );
  });
```

<img width="1288" alt="image" src="https://user-images.githubusercontent.com/31020403/222808271-f99fe065-8fe9-4fcc-a0e3-39c399bb7de1.png">

The nice part about this is that if you extract `gkeAppTypes` into the same helper that the component uses, adding a new `AppType` will also be covered by the tests.

One downside is that this seems difficult to maintain if the behavior starts to diverge.

I think what you have is ok as-is, too 👍

Reply:
oh that's nice, thanks